### ISSUE TITLE
Aliasing limitations on non-nullabilities constraints

### ISSUE THREAD

**Issue by bbenezech** [ts-safeql/safeql#273]
Hello dear maintainer,

This is what I have this morning with `@ts-safeql/eslint-plugin@3.4.5` and `libpg-query@15.0.2` on PG-14

`select count(1)` => `{ count: string }` ✅ 
`select count(1) as count` => `{ count: string }` ✅ 
`select count(1) as count2` => `{ count2: unknown }` 👻 

`select count(1)::int` => `{count: number}` ✅ 
`select count(1)::int as count` => `{count: number}` ✅ 
`select count(1)::int as count2` => `{count2: number | null}` 👻 

`select pid from pg_stat_activity` => `{ pid: number | null }` ✅ 
`select pid from pg_stat_activity where pid IS NOT NULL` => `{ pid: number }` ✅ 
`select pid from pg_stat_activity psa where pid IS NOT NULL` => `{ pid: number }` ✅ 
`select pid from pg_stat_activity psa where psa.pid IS NOT NULL` => `{ pid: number  | null }` 👻 
`select psa.pid from pg_stat_activity psa where psa.pid IS NOT NULL` => `{ pid: number  | null }` 👻 

Not exactly sure which failures are known limitations and which ones are unexpected? Would you like me to have a go at one of these cases?

**Comment by karlhorky** [comment]
I'm starting to confirm:

```ts
// 💥 Query has incorrect type annotation.
// 	Expected: { x: number }[]
//	  Actual: { x: number | null }[]
await sql<{ x: number }[]>`
  SELECT
    count(1)::int AS x
`;
```

**Comment by Newbie012** [comment]
fixed in [@ts-safeql/eslint-plugin@3.4.6](https://github.com/ts-safeql/safeql/releases/tag/%40ts-safeql%2Feslint-plugin%403.4.6)

**Comment by bbenezech** [comment]
I can confirm `select count(1)::int as count2` is now ✅ , but `select pid from pg_stat_activity psa where psa.pid IS NOT NULL` is still 👻 `{ pid: number  | null }`
